File: src/test/java/org/neo/gomina/model/inventory/FileInventoryTest.java
package org.neo.gomina.model.inventory;

import org.junit.Test;
import org.neo.gomina.model.inventory.file.FileInventory;

import java.util.Collection;
import java.util.List;

import static org.fest.assertions.Assertions.assertThat;

public class FileInventoryTest {

    @Test
    public void getEnvs() throws Exception {
        Inventory inventory = new FileInventory("data");
        Collection<Environment> environments = inventory.getEnvironments();
        System.out.println(environments);
        assertThat(environments).onProperty("id").containsOnly("PROD", "UAT", "DEV");
    }

    @Test
    public void getEnvironment() throws Exception {
        Inventory inventory = new FileInventory("data");
        Environment env = inventory.getEnvironment("UAT");
        System.out.println(env);
        assertThat(env.getName()).isEqualTo("tradex-uat");
        assertThat(env.getServices().size()).isGreaterThan(2);
    }

}

File: src/test/java/org/neo/gomina/model/inventory/FileInventoryTest.kt
package org.neo.gomina.model.inventory

import org.fest.assertions.Assertions.assertThat
import org.junit.Test
import org.neo.gomina.model.inventory.file.FileInventory
import java.io.File

class FileInventoryTest {

    @Test
    @Throws(Exception::class)
    fun getEnvs() {
        val inventory = FileInventory("data")
        val environments = inventory.getEnvironments()
        println(environments)
        assertThat(environments).onProperty("id").containsOnly("PROD", "UAT", "DEV")
    }

    @Test
    @Throws(Exception::class)
    fun getEnvironment() {
        val inventory = FileInventory("data")
        val env = inventory.getEnvironment("UAT")
        println(env)
        assertThat(env!!.name).isEqualTo("tradex-uat")
        assertThat(env!!.services.size).isGreaterThan(2)
    }

    @Test
    fun testReadOldModel() {
        val inventory = FileInventory("data")
        //Environment environment = inventory.readEnv(new File("env.monitorx-stg.json"));
        val environment = inventory.readEnv(File("env.tradex-uat.json"))
        println(environment)
    }
}